File: javascript/06-io-operations/read_input.js
// Reading a line from standard input
const readline = require('node:readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question("What is your name? ", (name) => {
  console.log(`Hello, ${name}! Welcome to JavaScript I/O.`);
  rl.close();
});

File: r/06-io-operations/read_input.R
# Reading from standard input
con <- file("stdin")

cat("Enter your name: ")
name <- readLines(con, n = 1)

cat("Enter your age: ")
age <- as.integer(readLines(con, n = 1))

close(con)

cat(sprintf("%s will be %d next year.\n", name, age + 1L))
